### PR TITLE
[kf5] upgrade to 5.89

### PR DESCRIPTION
**Pull request by wrobelda**
**Describe the pull request**
  Updates KF5 frameworks to 5.89.
  ~~Note this currently uses 5.89-rc1 tag. The release is [scheduled](https://community.kde.org/Schedules/Frameworks) for Dec 11th.~~ 5.89 was released.

- #### What does your PR fix?  

  Updates KF5 frameworks to 5.89.
  Removes some patches merged upstream.
  Also `dos2unix`-es some of the early portfiles for consistency and reduction of maintenance burden. Apart from updating version/SHA512, no other changes were made to portfiles.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  No change.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

**Comment by wrobelda** [comment]
@JonLiu1993 this is now ready for a review.

File: ports/ecm/portfile.cmake [hunk 1]
```diff
@@ -30,3 +30,5 @@ file(INSTALL "${SOURCE_PATH}/COPYING-CMAKE-SCRIPTS" DESTINATION "${CURRENT_PACKA
 
 # Allow empty include directory
 set(VCPKG_POLICY_EMPTY_INCLUDE_FOLDER enabled)
+
+
```
Comment (JonLiu1993) [inline ports/ecm/portfile.cmake:34]: @wrobelda ,Thanks for your pr, we should not add unnecessary spaces casually

File: ports/kf5archive/portfile.cmake [hunk 2]
```diff
@@ -1,40 +1,42 @@
...
-    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
-endif()
-
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
-
-file(INSTALL "${SOURCE_PATH}/LICENSES/" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright")
-
-
+vcpkg_from_github(
```
Comment (JonLiu1993) [inline ports/kf5archive/portfile.cmake:1]: Could you please keep the original format of EOL for this file, maybe LF?

File: ports/kf5attica/portfile.cmake [hunk 3]
```diff
@@ -1,31 +1,33 @@
...
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
-
-if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
-    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
-endif()
-
-file(INSTALL "${SOURCE_PATH}/LICENSES/" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright")
-
-
+vcpkg_from_github(
```
Comment (JonLiu1993) [inline ports/kf5attica/portfile.cmake:1]: same as above

File: ports/kf5completion/portfile.cmake [hunk 4]
```diff
@@ -1,31 +1,33 @@
...
-if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
-    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
-endif()
-
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
-
-file(INSTALL "${SOURCE_PATH}/LICENSES/" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright")
-
+vcpkg_from_github(
```
Comment (JonLiu1993) [inline ports/kf5completion/portfile.cmake:1]: Same as above

File: ports/kf5config/portfile.cmake [hunk 5]
```diff
@@ -1,50 +1,52 @@
...
-    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")	
-endif()
-
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
-
-file(INSTALL "${SOURCE_PATH}/LICENSES/" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright")
-
-
+vcpkg_from_github(
```
Comment (JonLiu1993) [inline ports/kf5config/portfile.cmake:1]: Same as above

File: ports/kf5coreaddons/portfile.cmake [hunk 6]
```diff
@@ -1,38 +1,40 @@
...
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
-
-if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")	
-    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
-endif()
-
-file(INSTALL "${SOURCE_PATH}/LICENSES/" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright")
-
+vcpkg_from_github(
```
Comment (JonLiu1993) [inline ports/kf5coreaddons/portfile.cmake:1]: Same as above

File: ports/kf5coreaddons/portfile.cmake [hunk 7]
```diff
@@ -1,38 +1,40 @@
...
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")	
+    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
+endif()
+
+file(INSTALL "${SOURCE_PATH}/LICENSES/" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright")
+
+
+
```
Comment (JonLiu1993) [inline ports/kf5coreaddons/portfile.cmake:40]: We should not add unnecessary spaces.

File: ports/kf5crash/portfile.cmake [hunk 8]
```diff
@@ -29,3 +29,5 @@ endif()
 file(INSTALL "${SOURCE_PATH}/LICENSES/" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright")
 
 
+
+
```
Comment (JonLiu1993) [inline ports/kf5crash/portfile.cmake:33]: We should not add unnecessary spaces

File: ports/kf5globalaccel/portfile.cmake [hunk 9]
```diff
@@ -1,35 +1,35 @@
...
- )
-
-if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
-    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
-endif()
-
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
-file(INSTALL "${SOURCE_PATH}/LICENSES/" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright")
+vcpkg_from_github(
```
Comment (JonLiu1993) [inline ports/kf5globalaccel/portfile.cmake:1]: Same as above

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
@wrobelda ,I reviewed the previous ports and found many common problems that need to be modified：
- Please keep the original format of EOL for this file, maybe LF
- Second: should not add unnecessary spaces in portfile.cmake

File: ports/ecm/portfile.cmake (hunk 1 above)
Comment (wrobelda) [reply ports/ecm/portfile.cmake:34]: Thanks. This is second time this happens, I believe https://github.com/microsoft/vcpkg/pull/19679 is causing this, which I use to prepare these updates.

File: ports/kf5archive/portfile.cmake (hunk 2 above)
Comment (wrobelda) [reply ports/kf5archive/portfile.cmake:1]: I noted that in my original comment:

> Also dos2unix-es some of the early portfiles for consistency and reduction of maintenance burden. Apart from updating version/SHA512, no other changes were made to portfiles.

This is for convenience – I am the maintainer of these files and update these on macOS and/or Linux, so having them conform to Unix standard is just less burden. As noted, no other changes were made and if possible, I'd like to have a permission to change the EOL of these few port files.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Stupid questions: Is there a reason kf5diagram is completely different to the other kf5 ports?

**Review by BillyONeal — APPROVED** [review]

**Pull request merged**